alphadia/workflow/peptidecentric.py

```diff
@@ -319,119 +314,85 @@ def get_batch_plan(self):
 
         return plan
 
-    def start_of_calibration(self):
-        self.batch_plan = self.get_batch_plan()
+    def extract_optimization_data(self, target, min_iterations):
+        """Search parameter optimization (i.e. refinement of tolerances for RT, MS2, etc.) is performed on a subset of the elution groups in the spectral library.
+        The number of elution groups which must be searched to get a sufficiently large number for robust calibration varies depending the library used and the data.
+        This function searches an increasing number of elution groups until a sufficient number (determined by target) of precursors are identified at 1% FDR.
+        It then returns the elution group indexes which will be used to find the data in the spectral library for search parameter optimization.
 
-    def start_of_epoch(self, current_epoch):
-        self.com.current_epoch = current_epoch
+        Parameters
+        ----------
 
-        # if self.neptune is not None:
-        #    self.neptune["eval/epoch"].log(current_epoch)
+        target : int
+            The number of precursors which must be identified at 1% FDR to stop the extraction.
+
+        """
 
         self.elution_group_order = self.spectral_library.precursor_df[
             "elution_group_idx"
         ].unique()
         np.random.shuffle(self.elution_group_order)
 
-        self.calibration_manager.predict(
-            self.spectral_library._precursor_df, "precursor"
-        )
-        self.calibration_manager.predict(self.spectral_library._fragment_df, "fragment")
-
-        # make updates to the progress dict depending on the epoch
-        if self.com.current_epoch > 0:
-            self.com.recalibration_target = self.config["calibration"][
-                "recalibration_target"
-            ] * (1 + current_epoch)
-
-    def start_of_step(self, current_step, start_index, stop_index):
-        self.com.current_step = current_step
-        # if self.neptune is not None:
-        #    self.neptune["eval/step"].log(current_step)
+        batch_plan = self.get_batch_plan()
 
-        #    for key, value in self.com.__dict__.items():
-        #        self.neptune[f"eval/{key}"].log(value)
+        features = []
+        fragments = []
+        for current_step, (start_index, stop_index) in enumerate(batch_plan):
+            self.reporter.log_string(
+                f"=== Step {current_step}, extracting elution groups {start_index} to {stop_index} ===",
+                verbosity="progress",
+            )
 
-        self.reporter.log_string(
-            f"=== Epoch {self.com.current_epoch}, step {current_step}, extracting elution groups {start_index} to {stop_index} ===",
-            verbosity="progress",
-        )
+            eg_idxes = self.elution_group_order[start_index:stop_index]
+            batch_df = self.spectral_library._precursor_df[
+                self.spectral_library._precursor_df["elution_group_idx"].isin(eg_idxes)
```

Comment:
This is just a comment/thought. I did this as a "cheap" way to get random elution groups from the library. In hindsight this operation is quite expensive for a large library. We can think of strategies how to do this different in the future.